### PR TITLE
Fix the flaky Netty Connection Tests

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
The reason for the test failure is that: after shutting down the server, we cannot start the server immediately.
Refactor the tests, remove duplicate code, useless tests and add more checks.
Addressed all java style issues.

### CODE REVIEW

File: pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java [hunk 1]
```diff
@@ -15,224 +15,101 @@
...
-      return _response;
-    }
-
-    public void setResponse(String response) {
-      _response = response;
-    }
+  private void closeClientConnection()
+      throws Exception {
+    _nettyTCPClientConnection.close();
+    Uninterruptibles.sleepUninterruptibly(100L, TimeUnit.MILLISECONDS);
```
Comment (mcvsubbu) [inline pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java]: please add a comment here as to why the sleep is needed (i.e. what event are you waiting for)

File: pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java [hunk 2]
```diff
@@ -15,224 +15,101 @@
...
-
-    @Override
-    public RequestHandler createNewRequestHandler() {
-      return _requestHandler;
-    }
-
+  private void closeServerConnection()
+      throws Exception {
+    _nettyTCPServer.shutdownGracefully();
+    Uninterruptibles.sleepUninterruptibly(1L, TimeUnit.SECONDS);
```
Comment (mcvsubbu) [inline pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java]: please add a comment as to what even the sleep is waiting for.

File: pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java]: Fixed

File: pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java (hunk 2 above)
Comment (Jackie-Jiang) [inline pinot-transport/src/test/java/com/linkedin/pinot/transport/netty/NettyCloseChannelTest.java]: Fixed

**Pull request merged**